### PR TITLE
importer: add TDE support for upload sst

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve?

Problem Summary:
Add TDE support for the `upload` method, so the ingest sst by tikv-importer should be encrypted.

### What is changed and how it works?

What's Changed:

### Check List 

Tests 

- Unit test
- Integration test

Side effects

### Release note 
- No release note

### CODE REVIEW

**Comment by glorv** [comment]
cc @yiwu-arbug  @hunterlxt

File: components/sst_importer/src/sst_importer.rs [hunk 1]
```diff
@@ -964,25 +999,37 @@ mod tests {
...
         {
-            let mut f = ImportFile::create(meta, path.clone()).unwrap();
+            let mut f = ImportFile::create(meta, path.clone(), data_key_manager.clone()).unwrap();
             f.append(data).unwrap();
             f.finish().unwrap();
             assert!(!path.temp.exists());
             assert!(path.save.exists());
+            if let Some(ref manager) = data_key_manager {
+                let info_res = manager.get_file(path.save.to_str().unwrap());
+                assert!(info_res.is_ok());
```
Comment (kennytm) [inline components/sst_importer/src/sst_importer.rs:1012]:
```suggestion
```

useless assert, `unwrap()` already asserts.

**Review by kennytm — COMMENTED** [review]
(rest) LGTM

File: components/sst_importer/src/sst_importer.rs [hunk 2]
```diff
@@ -724,20 +752,29 @@ impl ImportFile {
 
     pub fn finish(&mut self) -> Result<()> {
         self.validate()?;
-        self.file.take().unwrap().sync_all()?;
+        self.file.take().unwrap().flush()?;
```
Comment (hunterlxt) [inline components/sst_importer/src/sst_importer.rs:755]: Can you post the reference to illustrate `flush` has same semantic with `sync_all`

File: components/sst_importer/src/sst_importer.rs [hunk 3]
```diff
@@ -883,20 +916,20 @@ mod tests {
...
+            assert!(dir.create(&meta, None).is_err());
         }
 
         // Test ImportDir::delete()
         {
             File::create(&path.temp).unwrap();
             File::create(&path.save).unwrap();
             File::create(&path.clone).unwrap();
-            dir.delete(&meta).unwrap();
+            dir.delete(&meta, None).unwrap();
```
Comment (overvenus) [inline components/sst_importer/src/sst_importer.rs:932]: Please add a test to make sure delete works with TDE. Also please make sure key_manger does not contain a file that has been cleaned up.

File: components/sst_importer/src/sst_importer.rs [hunk 4]
```diff
@@ -698,21 +707,40 @@ impl fmt::Debug for ImportPath {
...
+        key_manager: Option<Arc<DataKeyManager>>,
+    ) -> Result<ImportFile> {
+        let file: Box<dyn Write + Send> = if let Some(ref manager) = key_manager {
+            // key manager will truncate existed file, so we should check exist manually.
+            if path.temp.exists() {
+                return Err(Error::Io(io::Error::new(
+                    io::ErrorKind::AlreadyExists,
+                    format!("file already exists, {}", path.temp.to_str().unwrap()),
+                )));
+            }
```
Comment (overvenus) [inline components/sst_importer/src/sst_importer.rs:728]: Does it break restoring from a checkpoint where there are some incomplete files left in TiKV nodes?

Comment (glorv) [reply components/sst_importer/src/sst_importer.rs:728]: The following original implementation with `File` returns an error if the file exists. 
See: https://github.com/tikv/tikv/blob/187774ed9433af80d117b6576179515b2e24ab6a/components/sst_importer/src/sst_importer.rs#L706-L717

**Review by yiwu-arbug — APPROVED** [review]
LGTM on my side, but I didn't look into importer logic.

File: components/sst_importer/src/sst_importer.rs (hunk 4 above)
Comment (kennytm) [reply components/sst_importer/src/sst_importer.rs:745]: Lightning never reuses the same SSTMeta name, even when using checkpoints.

**Review by overvenus — APPROVED** [review]
lgtm

**Pull request merged**